### PR TITLE
feat(linter): support class sorting for `tw.div`

### PR DESCRIPTION
**Pull request by Yz4230**
## Summary

This PR adds support for class sorting in the `tw.div` syntax, commonly used with libraries such as [twin.macro](https://github.com/ben-rogerson/twin.macro) and [react-twc](https://github.com/gregberge/twc). 

```diff
- tw.div`px-2·foo·p-4·bar`;
+ tw.div`foo·bar·p-4·px-2`;
```

To enable sorting for `tw.div` or similar patterns, need to include a configuration entry like `tw.*` in `options.functions`.

## Test Plan

- Updated snapshots for `useSortedClasses` tests to reflect the added support for the `tw.div` syntax.

### CODE REVIEW

**Review by ematipico — CHANGES_REQUESTED** [review]
Thank you for your contribution! Three things are missing in this PR: 
- A new changelog line that explains the new **feature**
- Update the documentation, and explain the new feature. You should add something like `**Since v2.0.0**`
- Update the documentation and add a new example with this new feature. The contribution guide [explains how to do so](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md#using-rule-options) 

Also, one question: what was the reason for adding `tw.*` not `tw.div`?

File: crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/options.rs [hunk 1]
```diff
@@ -33,6 +33,25 @@ impl UtilityClassSortingOptions {
...
+            if zip.all(|(m, p)| m == "*" || m == p)
+                && matcher.next().is_none()
+                && parts.next().is_none()
+            {
+                return true;
+            }
+        }
+
+        false
+    }
```
Comment (chansuke) [inline crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/options.rs:53]:
```suggestion
    pub(crate) fn match_function(&self, name: &str) -> bool {
        self.functions.iter().flatten().any(|matcher| {
            let mut matcher_parts = matcher.split('.');
            let mut name_parts = name.split('.').clone();

            let all_parts_match = matcher_parts
                .by_ref()
                .zip(name_parts.by_ref())
                .all(|(m, p)| m == "*" || m == p);

            all_parts_match
                && matcher_parts.next().is_none()
                && name_parts.next().is_none()
        })
    }
```
uses .any() to replace the explicit loop and manual return true. the result is functionally equivalent, with the same time and space complexity.
I prepared some testcases and  tested this on the playground, and the behavior matches the original implementation.

Comment (Yz4230) [reply crates/biome_js_analyze/src/lint/nursery/use_sorted_classes/options.rs]: Thank you for suggesting this!🥹
Committed without using redundant `.clone()`! [2e70bca](https://github.com/biomejs/biome/pull/4699/commits/2e70bca310e62695e8ff4c9bcf20890fa38afd7f)

```diff
-            let mut name_parts = name.split('.').clone();
+            let mut name_parts = name.split('.');
```

**Comment by Yz4230** [comment]
@ematipico 
Thank you very much for your review!
I added the changelog and updated the document.

> Also, one question: what was the reason for adding `tw.*` not `tw.div`?

This is for supporting tags other than `div` such as `tw.span` or `tw.a`.

**Review by ematipico — APPROVED** [review]

**Comment by Yz4230** [comment]
@ematipico 

Thank you for your review and approve!✨

Which version is this PR planned to be included in?
I marked it as v2.0.0 in the Files changed, but since there are no breaking changes, I think it could be released earlier.

If there are any shortcomings in this PR, I would appreciate it if you could point them out👀

**Comment by dyc3** [comment]
The next version we release will likely be 2.0. We aren't planning any minor releases before that.

**Comment by Yz4230** [comment]
Nice! Thank you!

**Pull request merged**